### PR TITLE
Set version to 0.5.0-develop

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by chriseth** [comment]
Tests are broken :)

**Comment by axic** [comment]
Need to also replace the the pragmas in the imports sources and solcjs.

**Comment by axic** [comment]
I think we can cherrypick and merge some of these commits.

**Comment by axic** [comment]
@chriseth I propose to merge 95e07bd and fcec6a4 in a separate PR right now.

**Comment by chriseth** [comment]
What are the PRs that contain these commits?

**Comment by axic** [comment]
This PR :)

**Comment by axic** [comment]
Moved them into #4625.

**Comment by axic** [comment]
Working on this PR now.

**Comment by axic** [comment]
@chriseth need to squash these things, but can you review the end result?

File: docs/abi-spec.rst [hunk 1]
```diff
@@ -468,7 +468,7 @@ For example,
 
 ::
 
-    pragma solidity >0.4.24;
+    pragma solidity >=0.4.24 <0.6.0;
```
Comment (axic) [inline docs/abi-spec.rst]: Need to fix this, I think this supposed to be a "0.5.0" check.

File: docs/050-breaking-changes.rst [hunk 2]
```diff
@@ -327,7 +327,7 @@ New version:
 
 ::
 
-   pragma solidity >0.4.25;
+   pragma solidity >=0.4.26 <0.6.0;
```
Comment (chriseth) [inline docs/050-breaking-changes.rst]: We actually could use something like `>0.4.99 <0.6.0`.

**Review by chriseth — DISMISSED** [review]

File: docs/050-breaking-changes.rst (hunk 2 above)
Comment (axic) [reply docs/050-breaking-changes.rst]: I was going to change the "0.5.0" ones to `>0.4.42 <0.6.0`, added as a distinct commit here.

Comment (chriseth) [reply docs/050-breaking-changes.rst]: I'm fine with whatever :)

**Comment by axic** [comment]
@chriseth this should be finally ready (also rebased on `develop` to ensure any docs update is captured) if you want to give it another glimpse

**Review by chriseth — APPROVED** [review]
Perhaps squash?

**Comment by axic** [comment]
Sorry, could have squashed the "Better regex" into the commit above, but left the other separate (especially the 0.5.0 specific commit) for traceability.

**Comment by lazaridiscom** [comment]
> for traceability

traceability is a good thing.

**Pull request merged**